### PR TITLE
fix concurrent map access in debug handlers

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/36448

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -302,7 +302,11 @@ func (s *DiscoveryServer) Syncz(w http.ResponseWriter, req *http.Request) {
 // Can be combined with the push debug interface to reproduce changes.
 func (s *DiscoveryServer) registryz(w http.ResponseWriter, req *http.Request) {
 	all := s.Env.ServiceDiscovery.Services()
-	writeJSON(w, all, req)
+	copied := make([]*model.Service, len(all))
+	for _, svc := range all {
+		copied = append(copied, svc.DeepCopy())
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/debug.go:307]: maybe it has been fixed by #39058, whatever we need a test

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:307]: I don't this it is fixed by that. The concurrent map write might still come if JSON encoder is encoding AddressMap while it is modified. There is already a test but it is difficult to create a test for this I guess. Any ideas?

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:307]: @hzxuzhonghu any thoughts?

Comment (hzxuzhonghu) [reply pilot/pkg/xds/debug.go:307]: I believe there is no mutating of `AddressMap` before deepcopy now. So this can not happen now.

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:307]: Ok. Cool I will close this

**Pull request closed without merging**